xilem_core/src/view.rs

```diff
@@ -224,7 +238,12 @@ where
         message: Message,
         app_state: &mut State,
     ) -> MessageResult<Action, Message> {
-        self.deref()
-            .message(view_state, id_path, message, app_state)
+        let message_result =
+            self.deref()
+                .message(&mut view_state.view_state, id_path, message, app_state);
+        if matches!(message_result, MessageResult::RequestRebuild) {
```

Comment:
Can you point towards the equivalent logic in Xilem Classic?

Just trying to ascertain if this is new semantics being added or just something I missed when porting this through.
Currently, I'm not aware of `MessageResult` having any semantics.

Reply:
> Just trying to ascertain if this is new semantics being added or just something I missed when porting this through.

I think missed? This is/was already in the `Memoize` view (now and then), and was used similarly in the `async_list` in the xilem_tokio branch in druid. It makes sense, since views don't have the same access in `View::message` to stuff as in `View::rebuild`, so they need to tell the parent views, via the `MessageResult`, that even though they don't seem to have changed, that they need a rebuild, so that the logic in `View::rebuild` can handle the rest. I guess it's related to [this comment](https://github.com/linebender/xilem/pull/456#issuecomment-2254721485)

Reply:
Yeah, fair enough. This does seem right.

It just seems a shame to lose the simplicity of the `Arc` view; I had been under the impression that the inner view would never be rebuilt if there was no change.

Reply:
Yeah I know what you mean, though I expect when we go to something like an `Arc` based signal/dirty-tracking system as described [here](https://xi.zulipchat.com/#narrow/stream/354396-xilem/topic/.60impl.20View.20for.20Arc.3CV.3A.20View.3E.60) (which I still haven't found a satisfying solution for though), that something like that would need to be done anyways.